### DISCUSSION TITLE
DatePicker week type

### DISCUSSION THREAD

**Discussion by martsim6** [rsuite/rsuite#3178] [Q&A] [answered]
Hello there, I would like to ask if there is any possibility to use single DatePicker with picking weeks only, something like:

`<DatePicker type='week' />`

I saw in [docs](https://rsuitejs.com/components/date-picker/#native-pickers) usage with native pickers as input, but I really need that DatePicker component to make it work with week type. Also in [RangeDatePicker](https://rsuitejs.com/components/date-picker/#native-pickers) there is `hoverRange` prop with 'week' option, but in single picker it's missing.

Is there any option to get it work in single picker now, or is it feature, which needs to be implemented yet?

Thanks

**Comment by simonguo** [comment, marked as answer]
Hi @martsim6  you can set showOneCalendar and showWeekNumbers on DateRangePicker to see if they meet your business requirements.
```tsx
<DateRangePicker oneTap showOneCalendar showWeekNumbers hoverRange="week" ranges={[]}/>
```

**Reply by martsim6** [reply]
Thanks, this did the thing. I missed `showOneCalendar` prop in range picker.

**Comment by SevenOutman** [comment]
Hi @martsim6, what's your expect output value of this `type="week"` variant? If it's a `[Date, Date]` tuple, DateRangePicker may fit your need. Or if you only need to pick one week number out of 52, I think a simple SelectPicker could be enough.

**Reply by martsim6** [reply]
Hello @SevenOutman , I would expect to ouput be the same as it is when you use classic `<input type="week" />`. It's `year-weekNumber`. Your solution is surely possible, but in this case you would need at least 2 selects, first for year pick second for week pick. And in this case, this is not what my company business wants, they want there a date picker, not selects :/
